terraso_backend/apps/project_management/graphql/projects.py

```diff
@@ -398,6 +398,10 @@ def mutate_and_get_payload(cls, root, info, project_id, user_id):
             cls.not_allowed(
                 MutationTypes.DELETE, msg="User not allowed to remove member from project"
             )
+
+        if project.is_sole_manager(user):
+            cls.not_allowed(MutationTypes.DELETE, msg="Cannot remove the last manager from project")
```

Comment:
question: should we use rules.test_rule(), like we do for landscape and groups? cc @josebui

Reply:
The enforcement mechanisms for validation are generally inconsistent across the mutation code and I did not want to introduce indirection + dependence on a library for an extremely straightforward check here. For example, just above this we have similar checks like `if not target_membership:` and `if not requester_membership:`.